Packs/BloodHoundEnterprise/Integrations/BloodHoundEnterprise/BloodHoundEnterprise_test.py

```diff
@@ -0,0 +1,397 @@
+from freezegun import freeze_time
+import demistomock as demisto
+from unittest.mock import Mock
+import json
+
+
+def load_test_events():
+    with open("test_data/event_list.json") as file:
+        return json.load(file)
+
+
+def create_mock_client(test_events):
+    client = Mock()
+
+    # Mock the search_events method to return slices of the test_events
+    def mock_search_events(limit, from_date, until_date, skip):
+        return test_events[skip: skip + limit]
+
+    client.search_events.side_effect = mock_search_events
+    return client
+
+
+@freeze_time("2024-11-22T13:28:27.698038Z")
+def test_fetch_events_first_time(mocker):
+    """
+    Given:
+        - A mock client with a predefined set of test events loaded from a JSON file.
+        - The current date and time are frozen at "2024-11-22T13:28:27.698038Z".
+        - No previous fetch history from `demisto.getLastRun` (indicating the first time fetching events).
+        - The parameter for `max_events_per_fetch` is set to "7".
+
+    When:
+        - The `fetch_events` function is called to retrieve events with the mock client.
+
+    Then:
+        - Ensure that exactly 7 events are fetched.
+        - Validate that the next run metadata (`next_run`) is correctly updated with:
+            - `last_event_created_at` timestamp.
+            - `last_event_id` of the last fetched event.
+            - `prev_fetch_id` set to 1.
+            - `skip` indicating the number of events fetched.
+        - Verify that the first event in the fetched list has the correct ID (2051).
+    """
+    from BloodHoundEnterprise import fetch_events
+
+    test_events = load_test_events()
+    client = create_mock_client(test_events)
+    mocker.patch.object(demisto, "getLastRun", return_value={})
+    mocker.patch.object(demisto, "debug")
+
+    next_run, events = fetch_events(
+        client=client,
+        params={"max_events_per_fetch": "7"},
+    )
+
+    assert len(events) == 7
+    assert next_run.get("last_event_created_at", "").startswith(
+        "2024-11-22T13:27:27.698038+"
```

Comment:
why not just equal?
Maybe add a new var `excepted_last_date`.

Reply:
This make a problems with the timezone.
I ran local with `== 2024-11-22T13:27:27.698038+02:00` and it passed
Then in the pre-commit it failed because it was equal to `2024-11-22T13:27:27.698038+00:00`